pkg/operator/ceph/cluster/osd/osd.go

```diff
@@ -314,11 +314,45 @@ func (c *Cluster) resolveNode(nodeName, deviceClass string) *cephv1.Node {
 	if rookNode == nil {
 		return nil
 	}
-	rookNode.Resources = k8sutil.MergeResourceRequirements(rookNode.Resources, cephv1.GetOSDResources(c.spec.Resources, deviceClass))
 
+	rookNode.Resources = c.getOSDResources(rookNode, rookNode.ResourceClassName, deviceClass)
 	return rookNode
 }
 
+func (c *Cluster) resourceRequirementFromResourceClass(resourceClassName string) (corev1.ResourceRequirements, error) {
```

Comment:
I need to create a map of the resource name and its limit as key, instead of calling this function, because it will call for-loop iteration again and again.

What will be a good place to initialize it?
I wonder if I can do it here https://github.com/rook/rook/blob/bb7306e6ee109775ed70e7b4239b5e93c22a8a33/pkg/operator/ceph/cluster/osd/osd.go#L211